core/sql/exp/exp_attrs.h

```diff
@@ -759,9 +763,10 @@ class SimpleType : public Attributes
   Int16 getScale()            {return scale_;}
   UInt16 getScaleAsUI()       {return (UInt16)scale_;}
 
-  Int32 getPrecision()        {return (UInt16)precision_;} // Treat as UInt16 in case 0x8000 or bigger
+  Int32 getPrecision()
+  {return (UInt16)precision_ | ((UInt32)precision_high_ << 16);}
   inline void setPrecision(Int32 precision)
-                              {precision_ = (Int16)precision;}
+  {precision_ = (Int16)precision; precision_high_ = (Int16)((UInt32)precision >> 16);}
```

Comment:
I would think it should be ok to change the precision to Int32 instead of keeping two different variables precision_ and precision_high_. Currently Trafodion doesn't support versioning. From this perspective, it is ok to change these structures which could be serialized across processes.

Reply:
Good to know that, thanks!